qiskit/providers/ibmq/managed/ibmqjobmanager.py

```diff
@@ -0,0 +1,196 @@
+# -*- coding: utf-8 -*-
+
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2019.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Job manager used to manage jobs for IBM Q Experience."""
+
+import logging
+from typing import List, Optional, Union
+from concurrent import futures
+
+from qiskit.circuit import QuantumCircuit
+from qiskit.pulse import ScheduleComponent, Schedule
+from qiskit.compiler import transpile
+
+from .exceptions import IBMQJobManagerInvalidStateError
+from .utils import format_job_details, format_status_counts
+from .managedjobset import ManagedJobSet, RetrievedManagedJobSet
+from ..ibmqbackend import IBMQBackend
+from ..accountprovider import AccountProvider
+
+logger = logging.getLogger(__name__)
+
+
+class IBMQJobManager:
+    """Job manager for IBM Q Experience."""
+
+    def __init__(self) -> None:
+        """Creates a new IBMQJobManager instance."""
+        self._job_sets = []  # type: List[ManagedJobSet]
+        self._executor = futures.ThreadPoolExecutor()
+
+    def run(
+            self,
+            experiments: Union[List[QuantumCircuit], List[Schedule]],
+            backend: IBMQBackend,
+            name: Optional[str] = None,
+            shots: int = 1024,
```

Comment:
The `JobManager.run()` method has been limited here in the options that it can accept. Is this intentional? Since these will be directly passed to the `qiskit.compiler.assemble()` function, why not allow all options that are allowed in assemble?

Reply:
The intent was to make the interface simpler. But you're right, people would want the `assemble` options. I added `**assemble_config` instead of listing them individually to avoid duplicating the options. However, this comes with the slight inconvenience to the user as they'd have to look up the `assemble()` options separately.

Reply:
The assemble function calls these `run_config`. Can we mirror that?
https://github.com/Qiskit/qiskit-terra/blob/812e213a012859693a3432becc377ace41e9034e/qiskit/assembler/assemble_circuits.py#L21

Also it might be a good idea to add them to the docstring here for reference.

Reply:
I added names of the options listed in `assemble()` to the docstring as examples, but I don't think it's necessary to duplicate their descriptions. Also would like to avoid having to keep them in sync.